CONTRIBUTING.md

```diff
@@ -0,0 +1,143 @@
+  should help with that.
+
+### Testing
+
+Tests are stored in [tests](/tests) directory.
+We use [Tox](https://pypi.org/project/tox) with configuration in [tox.ini](tox.ini).
+
+Running tests locally:
+```
+make prepare-check && make check
+```
+
+Running tests in a container (currently broken, PRs are welcome):
+```
+make build-tests && make test-in-container
+```
+
+As a CI we use [CentOS CI](https://ci.centos.org/job/packit-pr/) with a configuration in [Jenkinsfile](Jenkinsfile).
+
+### Makefile
+
+#### Requirements
+
+- [podman](https://github.com/containers/libpod)
+- [ansible-bender](https://pypi.org/project/ansible-bender)
+  - until [ansible](https://github.com/ansible/ansible/releases) 2.8 is released, it requires [buildah](https://github.com/containers/buildah)
+
+#### Targets
+
+Here are some important and useful targets of [Makefile](/Makefile):
+
+Use [ansible-bender](https://pypi.org/project/ansible-bender) to build container image from [recipe.yaml](recipe.yaml):
+```
+make build
+```
+
+Run [recipe-tests.yaml](recipe-tests.yaml) ansible playbook to install packages needed to run tests:
+```
+make prepare-check
+```
+
+Run tests locally:
+```
+make check
+```
+
+Start shell in a container from the image previously built with `make build`:
+```
+make shell
+```
+
+In a container, do basic checks to verify that packit can be distributed, installed and imported:
+```
+make check-pypi-packaging
+```
+
+### How to contribute code to packit
+
+1. Create a fork of the `packit` repository.
+2. Create a new branch just for the bug/feature you are working on.
+3. Once you have completed your work, create a Pull Request, ensuring that it meets the requirements listed below.
+
+### Requirements for Pull Requests
+
+* Please create Pull Requests against the `master` branch.
+* Please make sure that your code complies with [PEP8](https://www.python.org/dev/peps/pep-0008/).
+* One line should not contain more than 100 characters.
+* Make sure that new code is covered by a test case (new or existing one).
+* We don't like [spaghetti code](https://en.wikipedia.org/wiki/Spaghetti_code).
+* The tests have to pass.
+
+### Checkers/linters/formatters & pre-commit
+
+To make sure our code is compliant with the above requirements, we use:
+* [black code formatter](https://github.com/ambv/black)
+* [Flake8 code linter](http://flake8.pycqa.org)
+* [mypy static type checker](http://mypy-lang.org)
+
+There's a [pre-commit](https://pre-commit.com) config file in [.pre-commit-config.yaml](.pre-commit-config.yaml).
+To [utilize pre-commit](https://pre-commit.com/#usage), install pre-commit with `pip3 install pre-commit` and then either
+* `pre-commit install` - to install pre-commit into your [git hooks](https://githooks.com). pre-commit will now run all the checkers/linters/formatters on every commit.
+* Or if you want to manually run all the checkers/linters/formatters, run `pre-commit run --all-files`.
+
+
+Thank you!
```

Comment:
Nit: are these two lines necessary? Could we change it to "Thank you for your interest!".